src/ast/irbuilderbpf.cpp

```diff
@@ -672,5 +672,26 @@ void IRBuilderBPF::CreateSignal(Value *sig)
 #endif
 }
 
+void IRBuilderBPF::CreateFdPath(AllocaInst *path, Value *fd)
+{
+#ifdef HAVE_GET_FD_PATH
+  // int bpf_get_fdpath(char *path, u32 size, int fd)
+  // Return: 0 or error
+  FunctionType *fdpath_func_type = FunctionType::get(
+      getInt64Ty(), { getInt8PtrTy(), getInt32Ty(), getInt32Ty() }, false);
```

Comment:
```suggestion
      getInt64Ty(), { path->getType(), getInt32Ty(), getInt32Ty() }, false);
```

To avoid type issues.

Reply:
right, thanks